ui/js/exclude.js

```diff
@@ -369,6 +359,12 @@ jQuery(
 				$( '.stream-exclude-list tbody tr:not(.hidden) select.select2-select.ip_address', this ).each(
 					function() {
 						var firstSelected = $( 'option:selected', this ).first();
+
+						// Ugly hack to ensure we always pass an empty value or the order of rows gets messed up.
+						if ( ! firstSelected.length ) {
+							$( this ).append( '<option selected="selected"></option>' );
```

Comment:
This is the core fix -- we add a dummy option and mark it selected to ensure it gets delivered to the backend.